File: src/main/java/me/m64diamondstar/ingeniamccore/utils/messages/Colors.java
package me.m64diamondstar.ingeniamccore.utils.messages;

import net.md_5.bungee.api.ChatColor;
import org.bukkit.Bukkit;

import java.awt.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class Colors {

    private static Pattern pattern = Pattern.compile("#[a-fA-F0-9]{6}");

    public static String format(String msg) {
        Matcher match = pattern.matcher(msg);
        while(match.find()) {
            String color = msg.substring(match.start(), match.end());
            msg = msg.replace(color, ChatColor.of(color) + "");
            match = pattern.matcher(msg);
        }
        return ChatColor.translateAlternateColorCodes('&', msg).replace(":gs:", "✪");
    }

    public static Color getJavaColorFromString(String string){
        String[] args = string.split(", ");

        int r = 0;
        int g = 0;
        int b = 0;

        try{
            r = Integer.parseInt(args[0]);
            g = Integer.parseInt(args[0]);
            b = Integer.parseInt(args[0]);
        }catch (NumberFormatException e){
            Bukkit.getLogger().warning(e.getCause() + ": Error with converting String to Int in " + e.getClass());
        }

        return new Color(r, g, b);
    }

    public static String format(String msg, MessageType messageType) {
        return messageType + msg;
    }

}

File: src/main/java/me/m64diamondstar/ingeniamccore/utils/messages/Colors.kt
package me.m64diamondstar.ingeniamccore.utils.messages

import net.md_5.bungee.api.ChatColor
import java.awt.Color
import java.util.regex.Pattern

object Colors {
    private val pattern = Pattern.compile("#[a-fA-F0-9]{6}")
    @JvmStatic
    fun format(msg: String): String {
        var msg1 = msg
        var match = pattern.matcher(msg1)
        while (match.find()) {
            val color = msg1.substring(match.start(), match.end())
            msg1 = msg1.replace(color, ChatColor.of(color).toString() + "")
            match = pattern.matcher(msg1)
        }
        return ChatColor.translateAlternateColorCodes('&', msg1).replace(":gs:", "\uE016")
    }

    fun getJavaColorFromHex(hex: String): Color {
        var hexString = hex.lowercase()
        if (hexString.startsWith("#")) {
            hexString = hexString.substring(1)
        }
        val rgbValue: Int = hexString.toInt(16)
        return Color(rgbValue)
    }

    fun getJavaColorFromString(string: String): Color? {
        val args = string.split(", ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        return try {
            val r = args[0].toInt()
            val g = args[1].toInt()
            val b = args[2].toInt()
            Color(r, g, b)
        } catch (e: NumberFormatException) {
            null
        } catch (e: IllegalArgumentException) {
            null
        }
    }

    fun format(msg: String, messageType: MessageType): String {
        return messageType.toString() + msg
    }
}